### PR TITLE
igl | opengl | Add a new parameter for PlatformDevice::createTextureFromNativeDepth() to specify the desired depth textureFormat for OpenGLES

### PR DESCRIPTION
**Pull request by vinsentli**
Should use TextureFormat::S8_UInt_Z24_UNorm as the default depth stencil texture format for OpenGLES(iOS & Android)?
Because there is no stencil texture in the default depth texture now.

### CODE REVIEW

**Comment by corporateshark** [comment]
> This change is not a problem in itself; however, multiple internal clients may rely on previous behaviour. For example, it doubles the memory footprint for the default depth texture on iOS in OpenGL. We have to gate this change.
> 
> @corporateshark I can propose to use `#if defined(IGL_CMAKE_BUILD)` for changing default DS format. WDYT?

I think maintaining two diverging code paths can add extra costs in this case without any obvious benefits. Any app can explicitly choose a format it wants to use.

**Comment by vinsentli** [comment]
Perhaps it would be better to allow users to add a new parameter to specify the desired textureFormat.

```
std::shared_ptr<ITexture> PlatformDevice::createTextureFromNativeDepth(CAEAGLLayer* nativeDrawable,
                                                                       TextureFormat textureFormat,
                                                                       Result* outResult)
```

**Comment by corporateshark** [comment]
> Perhaps it would be better to allow users to add a new parameter to specify the desired textureFormat.

It sounds like a better solution.

**Comment by vinsentli** [comment]
I have add a new parameter “TextureFormat depthTextureFormat” for “PlatformDevice::createTextureFromNativeDepth()” on egl & ios.
I did not modify any other platforms(wgl, glx, ...).

File: shell/android/jni/TinyRenderer.cpp [hunk 1]
```diff
@@ -119,7 +119,7 @@ void TinyRenderer::render(float displayScale) {
   case BackendTypeID::GLES3: {
     auto* platformDevice = platform_->getDevice().getPlatformDevice<opengl::egl::PlatformDevice>();
     surfaceTextures.color = platformDevice->createTextureFromNativeDrawable(&result);
-    surfaceTextures.depth = platformDevice->createTextureFromNativeDepth(&result);
+    surfaceTextures.depth = platformDevice->createTextureFromNativeDepth(igl::TextureFormat::S8_UInt_Z24_UNorm, &result);
```
Comment (corporateshark) [inline shell/android/jni/TinyRenderer.cpp:122]: The default value should remain `TextureFormat::Z_UNorm24`.

File: shell/ios/ViewController.mm [hunk 2]
```diff
@@ -138,7 +138,7 @@ - (void)initRenderSessionController {
     IGL_ASSERT(platformDevice);
     return igl::SurfaceTextures{
         .color = platformDevice->createTextureFromNativeDrawable((CAEAGLLayer*)layer_, nullptr),
-        .depth = platformDevice->createTextureFromNativeDepth((CAEAGLLayer*)layer_, nullptr),
+        .depth = platformDevice->createTextureFromNativeDepth((CAEAGLLayer*)layer_, igl::TextureFormat::S8_UInt_Z24_UNorm, nullptr),
```
Comment (corporateshark) [inline shell/ios/ViewController.mm:141]: Same. Use the old value ` TextureFormat::Z_UNorm24`.

**Review by corporateshark — CHANGES_REQUESTED** [review]

File: shell/ios/ViewController.mm (hunk 2 above)
Comment (vinsentli) [reply shell/ios/ViewController.mm]: The ios old value is Z_UNorm16. 
Done.

**Comment by corporateshark** [comment]
@vinsentli [Android CI](https://github.com/facebook/igl/actions/runs/9691670033/job/26743576397?pr=134) is broken:

```
ld: error: undefined symbol: igl::vulkan::android::NativeHWTextureBuffer::NativeHWTextureBuffer(igl::vulkan::Device const&, igl::TextureFormat)
```

**Comment by vinsentli** [comment]
> @vinsentli [Android CI](https://github.com/facebook/igl/actions/runs/9691670033/job/26743576397?pr=134) is broken:
> 
> ```
> ld: error: undefined symbol: igl::vulkan::android::NativeHWTextureBuffer::NativeHWTextureBuffer(igl::vulkan::Device const&, igl::TextureFormat)
> ```

@corporateshark done

**Review by corporateshark — APPROVED** [review]

**Comment by corporateshark** [comment]
Please fix:
```
shell\windows\opengles\App.cpp:159:72: error: too few arguments to function call, expected 2, have 1
      auto depth = platformDevice->createTextureFromNativeDepth(nullptr);
                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~        ^
```

The title does not correspond to the update diff content. Please update.

**Comment by vinsentli** [comment]
> Please fix:
> 
> ```
> shell\windows\opengles\App.cpp:159:72: error: too few arguments to function call, expected 2, have 1
>       auto depth = platformDevice->createTextureFromNativeDepth(nullptr);
>                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~        ^
> ```
> 
> The title does not correspond to the update diff content. Please update.

@corporateshark done

**Pull request closed without merging**